src/metal.cs

```diff
@@ -497,6 +535,20 @@ partial interface MTLComputeCommandEncoder : MTLCommandEncoder {
 #endif
 		[Export ("setImageblockWidth:height:")]
 		void SetImageblock (nuint width, nuint height);
+
+		[Mac (10,14, onlyOn64: true), iOS (12,0), TV (12,0)]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("memoryBarrierWithScope:")]
+		void MemoryBarrier (MTLBarrierScope scope);
```

Comment:
Whats does the method do? Should we add a verb to the method  name?

Reply:
We could go `EncodeMemoryBarrier`.

Reply:
@spouliot ^ ?

Reply:
I like verbs but it's already named as such on [OpenGL](https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/memoryBarrier.xhtml) and in [.net](https://msdn.microsoft.com/en-us/library/system.threading.thread.memorybarrier(v=vs.110).aspx)

Reply:
So back to `MemoryBarrier`? I updated it now :P

Reply:
Yea the fact that it's named `MemoryBarrier` in .NET is enough to convince me we should stay consistent.